Makefile

```diff
@@ -1,7 +1,6 @@
-CC = gcc
-AR = ar
-override CFLAGS := -std=gnu99 -D_FILE_OFFSET_BITS=64 -Wall -Wextra -O3 -g -MMD $(CFLAGS)
-LDFLAGS =
+CC ?= gcc
+AR ?= ar
```

Comment:
Make doesn't overwrite user-defined variables, yes, but this makefile does.

Test case:

```sh
# initial file
$ echo -e 'CC = gcc' > Makefile
$ echo -e 'all:' >> Makefile
$ echo -e '\t@echo $(CC)' >> Makefile
# Current makefile
$ make
gcc
# Comment out CC = line
$ sed -i '1s/^/#/' Makefile
$ make
cc
# Restore to current makefile
$ sed -i 's/#//' Makefile
# Export clang as CC
$ export CC=clang
# Doesn't work
$ make
gcc
# Comment out that line → works
$ sed -i '1s/^/#/' Makefile
$ make
clang
```